File: Armadillo/IndexToCoor_3d.cpp
#include "DQMC.h"
int IndexToCoor_3d(int* x, int* y, int* z, int index, int NumInEdge)
{
	*x = index % NumInEdge;
	*y = ((index - *x) / NumInEdge) % NumInEdge;
	*z = (index - *x - *y * NumInEdge) / (NumInEdge * NumInEdge);
	return 0;
}

File: Test/HubbardAdvanced/dimension/IndexToCoor_3d.m
function [x,y,z] = IndexToCoor_3d(index,NumInEdge)
    x = mod(index-1,NumInEdge)+1;
    y = mod((index-x)/NumInEdge,NumInEdge)+1;
    z = (index - x - NumInEdge*(y-1))/NumInEdge^2+1;
end

